### PR TITLE
compose-switch is installed as a Linux "alternative" to docker-compose

### PR DESCRIPTION
**Pull request by ndeloof**
This PR:
- change installation process so we rely on Linux "alternatives" to replace docker-compose (v1)
- Drop "enabled|disable-v2" features from compose-switch as user can select implementation by `alternatives`
- include Dockerfile for ubuntu and centos to demonstrate installation process works as expected (TODO: automate testing)
- drop building Windows and OSX binaries (Docker Desktop uses its own flavor of the switch)
- drop e2e test used to validate enable/disable v2 had the expected effect

see https://github.com/docker/docker-ce-packaging/pull/553#issuecomment-916136917 for context
and https://docker.atlassian.net/browse/IL-584 (internal issue tracker)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,32 @@
 Compose Switch 
 --------------
 
-This is a thin wrapper to actually run either docker-compose v1 (python) or docker compose v2 (compose CLI plugin) depending on user's configuration
+Compose Switch is a replacement to the Compose V1 `docker-compose` (python) executable. It translate the command line info Compose V2 `docker compose` then run the latter.
```
Comment (mat007) [inline README.md:4]:
```suggestion
Compose Switch is a replacement to the Compose V1 `docker-compose` (python) executable. It translates the command line into Compose V2 `docker compose` then run the latter.
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,32 @@
...
+
+## installation
+
+We provide an install script for automated installtion:
```
Comment (mat007) [inline README.md:8]:
```suggestion
We provide a script for automated installation:
```

File: README.md [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+```console
+$ chmode +x /usr/local/bin/compose-switch
+```
+1. rename docker-compose if you already have it installed as `/usr/local/bin/docker-compose` 
+```console
+$ mv /usr/local/bin/docker-compose /usr/local/bin/docker-compose-v1
+```
+1. define an "alternatives" group for `docker-compose` command: 
+```console
+$ update-alternatives --install /usr/local/bin/docker-compose docker-compose /$(which docker-compose) 1
```
Comment (fredericdalleau) [inline README.md:30]: If you renamed `docker-compose` to  `docker-compose-v1` this should fail

File: README.md [hunk 4]
```diff
@@ -1,4 +1,32 @@
...
+```console
+$ curl -fL https://github.com/docker/compose-switch/releases/download/v1.0.1/docker-compose-linux-amd64 -o /usr/local/bin/compose-switch
+```
+1. make compose-switch executable
+```console
+$ chmode +x /usr/local/bin/compose-switch
+```
+1. rename docker-compose if you already have it installed as `/usr/local/bin/docker-compose` 
+```console
+$ mv /usr/local/bin/docker-compose /usr/local/bin/docker-compose-v1
```
Comment (fredericdalleau) [inline README.md:26]: Is it possible not to rename it?

File: centos-test.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+# install compose v1
+RUN curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+RUN chmod +x /usr/local/bin/docker-compose
+
+# install compose v2
+RUN mkdir -p /usr/local/lib/docker/cli-plugins
+RUN curl -L https://github.com/docker/compose/releases/download/v2.0.0-rc.3/docker-compose-linux-amd64 > /usr/local/lib/docker/cli-plugins/docker-compose
+RUN chmod +x /usr/local/lib/docker/cli-plugins/docker-compose
+
+
```
Comment (mat007) [inline centos-test.Dockerfile:20]: nit:
```suggestion

```

**Review by mat007 — COMMENTED** [review]
Maybe add an example of how to switch in the readme?

File: README.md (hunk 4 above)
Comment (ndeloof) [reply README.md:26]: we need to define alternative `docker-compose` command that will come first in `PATH`, and unfortunately `/usr/local/bin` comes first (we should not use /usr/local/sbin`)

Comment (fredericdalleau) [reply README.md:26]: Ha, this is not the one from the package, but one that could have been installed by the user

File: README.md (hunk 3 above)
Comment (fredericdalleau) [reply README.md]: It won't fail because the package if any would be on /usr/bin/docker-compose

Comment (fredericdalleau) [reply README.md]: what if the package is not installed?

File: README.md [hunk 6]
```diff
@@ -1,4 +1,32 @@
...
+
+### Manual installation
+
+1. download compose-switch binary for your architecture
+```console
+$ curl -fL https://github.com/docker/compose-switch/releases/download/v1.0.1/docker-compose-linux-amd64 -o /usr/local/bin/compose-switch
+```
+1. make compose-switch executable
+```console
+$ chmode +x /usr/local/bin/compose-switch
```
Comment (StefanScherer) [inline README.md:22]:
```suggestion
$ chmod +x /usr/local/bin/compose-switch
```

File: README.md (hunk 4 above)
Comment (ndeloof) [reply README.md:26]: yes, as documented by https://docs.docker.com/compose/install/

File: README.md [hunk 7]
```diff
@@ -1,4 +1,59 @@
...
+
+1. download compose-switch binary for your architecture
+```console
+$ curl -fL https://github.com/docker/compose-switch/releases/download/v1.0.1/docker-compose-linux-amd64 -o /usr/local/bin/compose-switch
+```
+2. make compose-switch executable
+```console
+$ chmod +x /usr/local/bin/compose-switch
+```
+3. rename `docker-compose` binary if you already have it installed as `/usr/local/bin/docker-compose`
```
Comment (fredericdalleau) [inline README.md:24]: If you installed docker-compose manually, you need to rename it.
If you installed install the docker-compose package from your distro, this step is not necessary

File: README.md [hunk 8]
```diff
@@ -1,4 +1,59 @@
...
+```console
+$ chmod +x /usr/local/bin/compose-switch
+```
+3. rename `docker-compose` binary if you already have it installed as `/usr/local/bin/docker-compose` 
+```console
+$ mv /usr/local/bin/docker-compose /usr/local/bin/docker-compose-v1
+```
+4. define an "alternatives" group for `docker-compose` command: 
+```console
+$ update-alternatives --install /usr/local/bin/docker-compose docker-compose <PATH_TO_DOCKER_COMPOSE_V1> 1
```
Comment (fredericdalleau) [inline README.md:30]: If you installed docker-compose manually, use the path of the renamed file
If you installed install the docker-compose package from your distro, use the patch of docker-compose

**Review by fredericdalleau — APPROVED** [review]
LGTM

**Review by ulyssessouza — APPROVED** [review]
LGTM

**Comment by slonopotamus** [comment]
But how now user switches between V1/V2 on Windows?

**Comment by slonopotamus** [comment]
It also doesn't work on every Linux distro, see #10.

**Comment by ndeloof** [comment]
> But how now user switches between V1/V2 on Windows?

Docker Desktop includes its own flavor of the Compose switch, so you can configure V1 vs V2 on settings. Otherwise, you can just select if docker-compose V1 is first in your PATH.

**Comment by slonopotamus** [comment]
Well, I thought the whole point why this project exists was to provide a way to switch between V1 and V2 (it even has *switch* in its name!). But switch that can no longer switch is kinda... not a switch at all anymore?

**Comment by ndeloof** [comment]
the name might be confusing, but the goal of this project is to make it easy to "_switch from V1 to V2_" by translating docker-compose V1 command into V2. This never has been intended to offer a "switch" to select V1 or V2 depending on your daily mood.

Suggestion to rely on for Linux alternatives was introduced late by a discussion with teammates on Linux packaging, this is not the core of this project.

Maybe we should rename as `compose-transition` or anything less confusing...

**Comment by ericb-summit** [comment]
you can safely assume `docker-compose` v1 won't get any further updates, so some translation tool like this is much better lesser evil than keeping old unsupported binaries lying around

**Pull request merged**